topics/reference_guide/text_selection.md

```diff
@@ -0,0 +1,55 @@
+In many cases this is sufficient to provide a good user experience.
+However, sometimes it’s advantageous to provide additional regions that the user may wish to be able to select when extending or shrinking a selection. 
+
+This EP has two methods that need to be implemented:
+
+1. The `canSelect(PsiElement)` method is called on each PSI element, starting from the element at the cursor and walking up each of its parents.
+   By returning `true` for a particular element, you can indicate that further text-ranges should be included for the PSI element.
+2. In the `select(PsiElement, CharSequence, int, Editor)` method, the specific text-ranges within the PSI element of interest are calculated and returned.
+
+### Example Use-Case
+
+A possible use-case for custom language developers is a function call `f(a, b)` where the function call node has its two arguments as children.
+If the cursor was, for example, at argument `a`, extending the selection would first select argument `a` itself and in the next step grow to cover the whole function call.
+However, you might want to select the list of all arguments as an intermediate step.
+This can be achieved by implementing this EP in the following way:
+
+1. Create a class that implements the `ExtendWordSelectionHandler` interface and register it as a `com.intellij.extendWordSelectionHandler` EP in your `plugin.xml`.
+2. The `canSelect(PsiElement)` method should return `true` for the function call node.
+   That indicates that `select(PsiElement, CharSequence, int, Editor)` will be called for the function-call node.
+3. When the `select` method is called, you can use the function call PSI element or the editor text to extract the text range that spans the arguments `a` and `b` and return it as `List<TextRange>` with one element.
+
+### Further Insight and Debugging
+
+Looking at other implementations can be an effective way to get a better understanding of how this EP works.
+To get further insight into this EP, you may want to take a look at [`DocTagSelectioner`](upsource:///java/java-impl/src/com/intellij/codeInsight/editorActions/wordSelection/DocTagSelectioner.java), an existing `ExtendWordSelectionHandler` in the IntelliJ Platform whose implementation is small and can be studied easily.
+It provides the ability to select tag names like `@param` in JavaDoc comments.
+
+There are also some important places in the IntelliJ Platform to add breakpoints during debugging.
+When _Extend Selection_ is called by the user, it is handled by [`SelectWordHandler`](upsource:///platform/lang-impl/src/com/intellij/codeInsight/editorActions/SelectWordHandler.java).
+The majority of the work, however, is then done inside [`SelectWordUtil`](upsource:///platform/lang-impl/src/com/intellij/codeInsight/editorActions/SelectWordUtil.java), where `processElement` checks which implementations of this EP apply for the current PSI element.
```

Comment:
method names should be noted with `()` behind the name

Reply:
I have added them, but the reason why I left them out is that I didn't want to write the whole signature of the method. And simply using `processElement()` would be wrong, since there is no such method that doesn't have any arguments..